common/include/opae/cxx/properties.h

```diff
@@ -38,28 +38,62 @@ namespace types {
 
 class token;
 
+/** Wraps an OPAE fpga_properties object.
+ *
+ * properties are information describing an
+ * accelerator resource that is identified by
+ * its token.
```

Comment:
Should we add something about properties also being used as a query for enumeration?